js_modules/dagster-ui/packages/ui-core/src/runs/useRunsForTimeline.tsx

```diff
@@ -524,13 +527,18 @@ export const useRunsForTimeline = ({
         // Only fetch ongoing runs once every 30 seconds
         (async () => {
           if (lastFetchRef.current.ongoing < Date.now() - 30 * 1000) {
-            await fetchOngoingRunsQueryData();
+            const promise = fetchOngoingRunsQueryData();
             lastFetchRef.current.ongoing = Date.now();
+            if (!ongoingRunsDataRef.current) {
+              // Only block "loading" if we haven't fetched ongoing runs before.
+              await promise;
+            }
```

Comment:
I think this could allow multiple calls to `fetchOngoingRunsQueryData` to happen concurrently if they take longer than 30s? Maybe that's not a concern?

It also seems like this means that the `refreshState` does not show "Refreshing..." even though a data fetch is in progress - Is that desirable here? (maybe?)

Reply:
Yeah it's desirable here imo

Reply:
Actually, I'll change approach slightly here so that we still show "Refreshing..."